warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryModelVisitor.java

```diff
@@ -395,20 +388,18 @@ protected JexlNode expandBinaryNodeFromModel(JexlNode node, Object data) {
          * need to ensure that if we create a logical structure ( such as an or ) -- each literal references a unique identifier from the right. Otherwise,
          * subsequent visitors will reference incorrection sub trees, and potentially negate the activity of the query model visitor
          */
-        Set<List<Object>> newSet = Sets.newHashSet(FluentIterable.from(product).transform(new ProductTransformer()));
+        Set<List<Object>> newSet = product.stream().map(new ProductTransformer()::apply).collect(Collectors.toSet());
```

Comment:
```suggestion
        Set<List<JexlNode>> newSet = product.stream().map(list -> list.stream().map(RebuildingVisitor::copy).collect(Collectors.toList()))
                        .collect(Collectors.toSet());
```

I like this change, but it had me wondering if we actually still needed the ProductTransformer.  After playing around with this a little, I settled on this implementation (which causes a ripple-effect of changes in a few places).

The transformer handled the case where one of the elements in the list is NOT a JexlNode.  However, after looking at all the ways we populate the 'left' and 'right' sets, we are only ever populating them with JexlNodes, so instead of typing everything as 'object', I typed everything as JexlNode, which simplified the ProductTransformer logic down to straight 'copy' on each sublist.  That's what is reflected here.  

This change led to changes method signature changes in JexlNodeFactory, namely:
1) createNodeTreeFromPairs - the pairs argument changed to Set<List<JexlNode>> pairs
2) buildUntypedBinaryNode - the left and right arguments changed to JexlNode

Both of those methods accept simple objects as inputs, but they don't actually work for anything that doesn't extend JexlNode, so it makes sense to me to be more specific with our typing.  I know that this isn't your code, and it may be a little outside the scope of this PR, but I think this would be a good change if you're up for it.

Reply:
@jwomeara I made the changes necessary to remove ProductTransformer.